### PR TITLE
Make anyone can register block syncer code to the Glueby::BlockSyncer

### PR DESCRIPTION
**Pull request by rantan**
See the example code in class document [here](https://github.com/chaintope/glueby/pull/66/files#diff-677ffeedbfa8faa3ad96039a5c2bd5de890672b61fe7d63629a99d1f87f7367a)

### CODE REVIEW

File: lib/glueby/block_syncer.rb [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  #     end
+  #   end
+  #
+  #   BlockSyncer.register_syncer(Syncer)
+  #
+  # @example Unregister the synchronous logic
+  #   BlockSyncer.unregister_syncer(Syncer)
+  #
+  class BlockSyncer
+    # @!attribute r height
```
Comment (azuchi) [inline lib/glueby/block_syncer.rb:22]:
```suggestion
    # @!attribute [r] height
```

File: spec/tasks/glueby/contract/block_syncer_spec.rb [hunk 2]
```diff
@@ -17,13 +42,12 @@
...
+      expect(rpc).to receive(:getrawtransaction).exactly(0).times
       expect(Glueby::AR::SystemInformation.synced_block_height.int_value).to eq(0)
       expect(Glueby::Internal::Wallet::AR::Utxo.count).to eq(1)
       subject.invoke
       expect(Glueby::AR::SystemInformation.synced_block_height.int_value).to eq(2)
       expect(Glueby::Internal::Wallet::AR::Utxo.count).to eq(2)
     end
   end
-
 end
```
Comment (azuchi) [inline spec/tasks/glueby/contract/block_syncer_spec.rb:53]: The path to this file should also be `spec/tasks/glueby/block_syncer`.

**Review by azuchi — CHANGES_REQUESTED** [review]
In addition to this change,  following change are required:

* remove `wallet_adaptor.rake`. This is same with `ActiveRecordWalletAdapter::Syncer`, right?
* need to migrate `timestam.rake`.

**Comment by rantan** [comment]
> remove wallet_adaptor.rake. This is same with ActiveRecordWalletAdapter::Syncer, right?

It is almost the same but ActiveRecordWalletAdapter::Syncer is not a rake task.
How about modifying that the wallet_adapter.rake uses ActiveRecordWalletAdapter::Syncer internally. 
I wonder if the wallet_adapter rake task has use cases. I actually don't know. If not, removing wallet_adapter.rake is good for me.

**Comment by rantan** [comment]
@azuchi 

> need to migrate timestam.rake.

You mean glueby:contract:timestamp:confirm should be abrogated and the confirmation should be checked as a block syncer logic.

**Comment by rantan** [comment]
* I will remove wallet_adapter.rake
* the confirmation should be checked as a block syncer logic.
* Add README.md documentation how to use BlockSyncer in applications
* Update initilalizer generator to the latest specs.

**Comment by rantan** [comment]
I've updated all the above things.

File: lib/glueby.rb [hunk 3]
```diff
@@ -1,6 +1,12 @@
 require "glueby/version"
 require 'tapyrus'
 
+begin
+  require 'rails'
```
Comment (azuchi) [inline lib/glueby.rb:5]: Why require `rails`? It doesn't seem to be defined in the gemspec.

Comment (rantan) [reply lib/glueby.rb:5]: I was thinking that it is necessary if glueby is loaded in a rails project. Glueby needs to use Rails::Railtie module. However, I didn't confirm whether it causes some error or not. I will check it.

Comment (rantan) [reply lib/glueby.rb]: I removed these codes. There is no error if it is removed.

**Review by azuchi — APPROVED** [review]

**Pull request merged**